tests/general/views/test_consolidated_profile_view.py

```diff
@@ -0,0 +1,566 @@
+        ]
+
+    def test_metric_sibling_with_multiversion(self):
+        version2 = VersionFactory()
+        geo3 = self.root.add_child(code="GT")
+        self.create_boundary(geo3, version2)
+        self.create_boundary(self.geo1, version2)
+        configuration = self.hierarchy.configuration
+        configuration["versions"] = [self.version.name, version2.name]
+        self.hierarchy.configuration = configuration
+        self.hierarchy.save()
+
+        # self.version data
+        data = {
+            "geo1": [{"gender": "female", "age": "16", "count": 2}],
+            "geo2": [{"gender": "female", "age": "16", "count": 4}],
+            "geo3": [{"gender": "female", "age": "16", "count": 6}],
+        }
+
+        dataset1 = self.create_dataset(self.profile, self.version)
+        dataset2 = self.create_dataset(self.profile, version2)
+        for dataset in [dataset1, dataset2]:
+            self.multiple_datasetdata(dataset, self.geo1, data["geo1"])
+            self.multiple_datasetdata(dataset, self.geo2, data["geo2"])
+            self.multiple_datasetdata(dataset, geo3, data["geo3"])
+            indicator = self.create_indicator(dataset, "gender")
+            self.create_key_metrics(
+                self.profile, indicator, "female", "metric", "sibling"
+            )
+
+        response = self.get(
+            'all-details',
+            profile_id=self.profile.pk,
+            geography_code=self.geo1.code,
+            data={
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        metrics = response.data["profile"]["profile_data"]["category-female"]["subcategories"]["subcategory-female"]["key_metrics"]
+        assert metrics == [
+            {'label': 'metric', 'value': 0.3333333333333333, 'method': 'sibling'}
+        ]
+
+        # Request with version2
+        response = self.get(
+            'all-details',
+            profile_id=self.profile.pk,
+            geography_code=self.geo1.code,
+            data={
+                'version': version2.name,
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        metrics = response.data["profile"]["profile_data"]["category-female"]["subcategories"]["subcategory-female"]["key_metrics"]
+        assert metrics == [
+            {'label': 'metric', 'value': 0.25, 'method': 'sibling'}
+        ]
+
+
+class TestParentLayersData(ConsolidatedProfileViewBase):
+
+    def test_version_parent_and_siblings(self):
+        version1, geography, hierarchy, profile = self.create_base_data(
+            "v2", None, self.hierarchy, self.profile
+        )
+
+        data = [
+            {"gender": "male", "age": "15", "count": 1},
+            {"gender": "female", "age": "16", "count": 2},
+        ]
+        indicator = self.create_versioned_data(
+            version1, geography, profile, data=data
+        )
```

Comment:
why are we creating an indicator for a test that doesn't care about indicators or indicator data?